src/mesh/region_set.cc

```diff
@@ -0,0 +1,291 @@
+		if ( reg.is_valid() ) region_id = reg.id();
+		else THROW( ExcNonexistingLabel() << EI_Region_Label(region_label) );
+	}
+
+	this->add_region(mesh, region_id, region_label);
+
+    Input::Array element_list;
+	if (rec.opt_val("element_list", element_list) ) {
+		std::vector<unsigned int> element_ids;
+		for (Input::Iterator<unsigned int> it_element = element_list.begin<unsigned int>();
+				it_element != element_list.end();
+		        ++it_element) {
+			element_ids.push_back( *it_element );
+
+		}
+		mesh->modify_element_ids(region_id, element_ids);
+	}
+}
+
+
+
+const IT::Record & RegionSetFromElements::get_region_input_type()
+{
+    return IT::Record("From_Elements", "Region declared by name and enum of elements.")
+        .derive_from(RegionSetBase::get_input_type())
+		.declare_key("name", IT::String(), IT::Default::obligatory(),
+				"Label (name) of the region. Has to be unique in one mesh.\n")
+		.declare_key("id", IT::Integer(0), IT::Default::optional(),
+				"The ID of the region to which you assign label.")
+		.declare_key("element_list", IT::Array( IT::Integer(0) ), IT::Default::optional(),
+				"Specification of the region by the list of elements. This is not recomended")
+		.close();
+}
+
+
+
+const int RegionSetFromElements::registrar =
+		Input::register_class< RegionSetFromElements, const Input::Record &, Mesh * >("From_Elements") +
+		RegionSetFromElements::get_region_input_type().size();
+
+
+
+/*******************************************************************
+ * implementation of RegionSetBoundary
+ * Need new implementation, will be solved later.
+ */
+
+// RegionSetBoundary::RegionSetBoundary(const Input::Record &rec, Mesh *mesh) {}
+
+// const IT::Record & RegionSetBoundary::get_region_input_type() {}
+
+//const int RegionSetBoundary::registrar =
+//		Input::register_class< RegionSetBoundary, const Input::Record &, Mesh * >("Region_Boundary") +
+//		RegionSetBoundary::get_region_input_type().size();
+
+
+
+/*******************************************************************
+ * implementation of RegionSetUnion
+ */
+
+RegionSetUnion::RegionSetUnion(const Input::Record &rec, Mesh *mesh)
+{
+	string set_name = rec.val<string>("name");
+	Input::Iterator<Input::Array> labels = rec.find<Input::Array>("region_labels");
+
+	pair<string,string> set_names = mesh->region_db().get_and_check_operands(*labels);
+	RegionSet region_set = mesh->region_db().union_sets( set_names.first, set_names.second );
+	add_set(mesh, set_name, region_set);
+}
+
+
+
+const IT::Record & RegionSetUnion::get_region_input_type()
+{
+    return IT::Record("Union", "Defines region as a union of given two or more regions.")
+        .derive_from(RegionSetBase::get_input_type())
+		.declare_key("name", IT::String(), IT::Default::obligatory(),
+				"Label (name) of the region. Has to be unique in one mesh.\n")
+		.declare_key("region_labels", IT::Array( IT::String(), 2,2 ), IT::Default::obligatory(),
```

Comment:
Rename "region_labels" to "regions", array of size >=1.
Add key "region_ids".
Create union of all referenced regions.